src/lib/isPromise.js

```diff
@@ -0,0 +1,6 @@
+// an object is a promise if it has a `.then` function
+// https://github.com/then/is-promise/blob/master/index.js
+export default obj =>
+  !!obj &&
+  (typeof obj === 'object' || typeof obj === 'function') &&
+  typeof obj.then === 'function';
```

Comment:
How can `obj` both be of type `function` _and_ have `obj.then`?

Reply:
classes are just functions and functions can have properties, love javascript. this is just is-promise, btw — not my doing.